jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletWrapperTest.java

```diff
@@ -0,0 +1,151 @@
+    public void startServer() throws Exception
+    {
+        server = new Server();
+
+        localConnector = new LocalConnector(server);
+        server.addConnector(localConnector);
+
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+
+        ServletHolder servletHolder = context.addServlet(HelloServlet.class, "/hello");
+        servletHolder.setAsyncSupported(false);
+        FilterHolder filterHolder = context.addFilter(WrapFilter.class, "/*", EnumSet.of(DispatcherType.REQUEST));
+        filterHolder.setAsyncSupported(true);
+
+        HandlerList handlers = new HandlerList();
+        handlers.addHandler(context);
+        handlers.addHandler(new DefaultHandler());
+
+        server.setHandler(handlers);
+        server.start();
+    }
+
+    @AfterEach
+    public void stopServer()
+    {
+        LifeCycle.stop(server);
+    }
+
+    @Test
+    public void testWrapper() throws Exception
+    {
+        StringBuilder req = new StringBuilder();
+        req.append("GET /hello HTTP/1.1\r\n");
+        req.append("Host: local\r\n");
+        req.append("Connection: close\r\n");
+        req.append("\r\n");
+
+        String rawResponse = localConnector.getResponse(req.toString());
+        HttpTester.Response resp = HttpTester.parseResponse(rawResponse);
+        assertThat("Response.status", resp.getStatus(), is(200));
+    }
+
+    public static class HelloServlet extends HttpServlet
+    {
+        @Override
+        protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+        {
+            resp.setContentType("text/plain");
+            resp.setCharacterEncoding("utf-8");
+            resp.getWriter().println("Hello Test");
+        }
+    }
+
+    public static class WrapFilter implements Filter
+    {
+        @Override
+        public void init(FilterConfig filterConfig) throws ServletException
+        {
+            // ignore
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException
+        {
+            if (!(request instanceof HttpServletRequest) || !(response instanceof HttpServletResponse))
+            {
+                chain.doFilter(request, response);
+                return;
+            }
+
+            HttpServletRequest httpRequest = (HttpServletRequest)request;
+            HttpServletResponse httpResponse = (HttpServletResponse)response;
+
+            httpRequest = new NoopRequestWrapper(httpRequest);
+            chain.doFilter(httpRequest, httpResponse);
+        }
+
+        @Override
+        public void destroy()
+        {
+            // ignore
+        }
+    }
+
+    public static class NoopRequestWrapper extends HttpServletRequestWrapper
```

Comment:
Should probably rename this to something other then "Noop", perhaps "SimpleRequestWrapper"?